src/com/palmergames/bukkit/towny/event/actions/TownyExplodingBlockEvent.java

```diff
@@ -0,0 +1,150 @@
+package com.palmergames.bukkit.towny.event.actions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Part of the API which lets Towny's war and other plugins modify Towny's
+ * plot-permission-decision outcomes.
+ * 
+ * TownyExplodingBlockEvents are thrown when an explosion occurs 
+ * in a Towny world, causing block damage.
+ * 
+ * <br> - When a bed/respawn anchor explodes causing block damage.
+ * <br> - When a an entity explodes causing block damage.
+ * 
+ * @param vanillaBlockList - List of Blocks which were involved in the original explosion.
+ * @param townyFilteredList - List of Blocks Towny has already filtered, these blocks will explode unless {@link #setBlockList(List)} is used.
+ * @param mat - Material which caused the block explosion or null if it is an entity explosion.
+ * @param entity - Entity which caused the entity explosion or null if it is a block explosion. 
+ * 
+ * @author LlmDl
+ */
+public class TownyExplodingBlockEvent extends Event {
+
+	private static final HandlerList handlers = new HandlerList();
+	private final List<Block> vanillaBlockList;
+	private final List<Block> filteredBlockList;
+	private final Material material;
+	private final Entity entity;
+	private List<Block> blockList = new ArrayList<Block>();
+	private boolean isChanged = false;
+
+	/**
+	 * Event thrown when blocks are exploded by a block or entity.
+	 * 
+	 * @param vanillaBlockList - List of Blocks which were involved in the original explosion.
+	 * @param townyFilteredList - List of Blocks Towny has already filtered, these blocks will explode unless {@link #setBlockList(List)} is used.
+	 * @param mat - Material which caused the block explosion or null if it is an entity explosion.
+	 * @param entity - Entity which caused the entity explosion or null if it is a block explosion. 
+	 */
+	public TownyExplodingBlockEvent(List<Block> vanillaBlockList, List<Block> townyFilteredList, Material mat, Entity entity) {

```

Comment:
Nitpick, but maybe should be called `Blocks` event instead of `Block` event since it's more than one block.